src/inference/src/ie_core.cpp

```diff
@@ -494,13 +495,56 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
             res = LoadNetworkFromCache(cacheManager, hash, plugin, parsed._config, context, loadedFromCache);
             if (!loadedFromCache) {
                 res = compile_model_impl(network, plugin, parsed._config, context, hash);
+            } else {
+                updateCNNInfo(res, network);
             }
         } else {
             res = compile_model_impl(network, plugin, parsed._config, context, {});
         }
         return res;
     }
 
+    static ie::InputsDataMap copyInfo(const ie::InputsDataMap& networkInputs) {
+        ie::InputsDataMap _networkInputs;
+        for (const auto& it : networkInputs) {
+            ie::InputInfo::Ptr newPtr;
+            if (it.second) {
+                newPtr = std::make_shared<ie::InputInfo>();
+                newPtr->getPreProcess() = it.second->getPreProcess();
+                newPtr->setInputData(std::make_shared<ie::Data>(*it.second->getInputData()));
+            }
+            _networkInputs.emplace(it.first, newPtr);
+        }
+        return _networkInputs;
+    }
```

Comment:
can we reuse https://github.com/openvinotoolkit/openvino/blob/master/src/inference/dev_api/cpp_interfaces/interface/ie_iplugin_internal.hpp#L77-L85 here w/o introduction of new methods?